File: compute_fcst_full_dateRange_multi_us.pl
#!/usr/bin/perl

use File::Basename;
use strict;


my $DEBUG=0;

($#ARGV+2)==5 || die
"Usage: compute_fcst_full_dateRange_multi_us.pl list_sym startDate endDate N(8)
       Compute historical fcsts_full (all inds) for a given symbol list +dateRange
       It will process one symbol at a time;
       for each symbol, the dates are broken into N batches,
       and each batch is one thread.\n";

my $filename=$ARGV[0]; # portara text file


my $startDate=$ARGV[1];
my $endDate=$ARGV[2];
my $N=$ARGV[3];


# create blocks
my $NFold=$N;


open(INFILE, "$filename") || die "Couldn't open $filename: $!\n";

my @line; 
# loop each symbol
while(<INFILE>)
{
    #chomp the new line at the end
    chomp($_);
    @line =split;

    my $sym=$line[0];



    my $cmd1="
      compute_fcst_full_dateRange_singleSymManager_us.pl $sym $startDate $endDate $N";

    print "$cmd1\n";
    system("$cmd1");
}
close(INFILE);









__END__


my $pid = open my $fh,"-|","sleep 3";
print waitpid(28779,0); # Some other process
print waitpid($pid,0);


--cmd:
time cat /mnt/wbox1/portara/Futures/Continuous\ Contracts/Intraday\ Database/1\ Minute\ 24Hr/EU.001 | sed s/,/\ /g|fgrep -v DATE|gawk '{if($1>=20150115 && $1<=20150115){print $1,$2,$7}}' > /tmp/tmpVol.txt
# 35 sec
cat /tmp/tmpVol.txt|myPortaraAddMissingMinutes.pl 1 2 3|mygetcols.pl 1 2 6 >/tmp/tmpVolNorm.txt 
timesXmgr.pl /tmp/tmpVolNorm.txt 2 0|mygetcols.pl 1 4 2 3  >/tmp/tmpForplot.txt
xmgrByTime /tmp/tmpForplot.txt&

File: compute_fcst_full_dateRange_multi_us.py
#!/usr/bin/env python3

import sys
import subprocess

DEBUG = 0

# Check if the correct number of arguments is provided
if len(sys.argv) != 5:
    print("Usage: compute_fcst_full_dateRange_multi_us.py list_sym startDate endDate N(8)")
    print("Compute historical fcsts_full (all inds) for a given symbol list + dateRange")
    print("It will process one symbol at a time; for each symbol, the dates are broken into N batches, and each batch is one thread.")
    sys.exit(1)


# Get command-line arguments
filename = sys.argv[1]  # list of symbols from portara text file
start_date = sys.argv[2]
end_date = sys.argv[3]
N = sys.argv[4]  # Number of batches/threads

# Try to open the symbol list file
try:
    with open(filename, 'r') as infile:
        # Loop through each line in the file
        for line in infile:
            # Remove any trailing newlines or whitespaces
            line = line.strip()

            # Get the symbol from the line
            sym = line.split()[0]

            # Construct the command to execute
            cmd1 = f"compute_fcst_full_dateRange_singleSymManager_us.pl {sym} {start_date} {end_date} {N}"

            # Print and execute the command
            print(cmd1)
            subprocess.run(cmd1, shell=True)

except FileNotFoundError:
    print(f"Couldn't open {filename}: File not found.")
    sys.exit(1)


